src/api/cluster/pke/pkeservice/service.go

```diff
@@ -0,0 +1,269 @@
+)
+
+type (
+	Cluster       = cluster.Cluster
+	Process       = process.Process
+	ProcessEvent  = process.ProcessEvent
+	ProcessStatus = process.ProcessStatus
+)
+
+// +testify:mock:testOnly=true
+type logger logur.Logger
+
+// serviceError is an error that can be shown to the user and be marshalled (for cadence)
+type serviceError struct {
+	error
+	Message string
+}
+
+func (*serviceError) ServiceError() bool {
+	return true
+}
+
+func wrapServiceError(err error) serviceError {
+	return serviceError{err, err.Error()}
+}
+
+func newServiceError(text string, args ...interface{}) serviceError {
+	return wrapServiceError(errors.Errorf(text, args...))
+}
+
+type NodeStatus struct {
+	// name of node
+	Name string
+
+	// name of nodepool
+	NodePool string
+
+	// ip address of node (where the other nodes can reach it)
+	Ip string
+
+	// detailed description about the current bootstrapping status (including the cause of the failure)
+	Message string
+
+	// the current phase of the bootstrap process
+	Phase string
+
+	// if this is the final status report, that describes the conclusion of the whole process
+	Final bool
+
+	Status ProcessStatus
+
+	// exact time of event
+	Timestamp time.Time
+
+	// ID of the process registered earlier (register new process if empty)
+	ProcessID string
+}
+
+const (
+	Running  ProcessStatus = "running"
+	Failed   ProcessStatus = "failed"
+	Finished ProcessStatus = "finished"
+	Canceled ProcessStatus = "canceled"
+)
+
+// +kit:endpoint:errorStrategy=service
+// +testify:mock
+
+// Service provides an interface to PKE specific operations (i.e. [currently some of] those called by the pke installer)
+type Service interface {
+	// RegisterNodeStatus registers status reported by a node
+	RegisterNodeStatus(ctx context.Context, clusterIdentifier cluster.Identifier, nodeStatus NodeStatus) (resp RegisterNodeStatusResponse, err error)
+}
+
+type RegisterNodeStatusResponse struct {
+	ProcessID string
+}
+
+type service struct {
+	clusters    Store
+	processes   processService
+	idGenerator idGenerator
+	logger      logger
+}
+
+// +testify:mock:testOnly=true
+type idGenerator interface {
+	New() string
+}
+
+type uuidGenerator struct{}
+
+func (uuidGenerator) New() string {
+	return uuid.Must(uuid.NewV4()).String()
+}
+
+func nodeBRN(organizationID uint, clusterID uint, hostname string) brn.ResourceName {
+	id := path.Join(fmt.Sprint(clusterID), hostname)
+	return brn.New(organizationID, brn.NodeResourceType, id)
+}
+
+func clusterBRN(organizationID uint, clusterID uint) brn.ResourceName {
+	return brn.New(organizationID, brn.ClusterResourceType, fmt.Sprint(clusterID))
+}
+
+func brnProcess(p Process, brn brn.ResourceName) Process {
+	p.ResourceId = brn.ResourceID
+	p.OrgId = int32(brn.OrganizationID)
+	p.ResourceType = brn.ResourceType
+	return p
+}
+
+func (s service) RegisterNodeStatus(ctx context.Context, clusterIdentifier cluster.Identifier, nodeStatus NodeStatus) (resp RegisterNodeStatusResponse, err error) {
```

Comment:
Report?

Reply:
no, this service registers the report :) the http endpoint is used by pke to report the status

Reply:
I think the term is debatable. Report as an operation name should be fine. Save would also work.

Reply:
I don't like `Save`, nor Report for this.

Reply:
Well, I don't like Register. 🙂